cluster/src/main/scala/io/snappydata/impl/LeadImpl.scala

```diff
@@ -368,6 +373,43 @@ class LeadImpl extends ServerImpl with Lead
     LeadImpl.clearInitializingSparkContext()
   }
 
+  def configureAuthenticatorForSJS(): Unit = {
+    if (Misc.isSecurityEnabled) {
+      logInfo("Configuring authenticator for Snappy Job users.")
+      SnappyAuthenticator.auth = new SnappyAuthenticator {
+
+        import scala.concurrent.ExecutionContext.Implicits.global
+
+        override def authenticate(userPass: Option[UserPass]): Future[Option[AuthInfo]] = {
+          Future { checkCredentials(userPass) }
+        }
+
+        def checkCredentials(userPass: Option[UserPass]): Option[AuthInfo] = {
+          userPass match {
+            case Some(u) => {
+              try {
+                val props = new Properties()
+                props.setProperty(Attribute.USERNAME_ATTR, u.user)
+                props.setProperty(Attribute.PASSWORD_ATTR, u.pass)
+                val result = FabricDatabase.getAuthenticationServiceBase.authenticate(Misc
+                    .getMemStoreBooting.getDatabaseName, props)
+                if (result != null) {
+                  logInfo(s"ACCESS DENIED, user [${u.user}]. $result")
```

Comment:
Add the tag used on store side throughout: GfxdConstants.TRACE_AUTHENTICATION. This will ensure a clean way to filter out auth related logging and create an audit log. Later we can also add support in our logger to send these tags to a separate file if required.